### PR TITLE
Propagate labels on go_repo rule to subrepo targets (part 1)

### PR DESCRIPTION
**Pull request by mspraggs**
This will ensure that any labels passed to `go_repo` will be propagated to the various BUILD files produced by `please_go generate`. In turn, those labels will appear in the output of `plz query graph` when invoked on the root repo.

### CODE REVIEW

File: build_defs/go.build_defs [hunk 1]
```diff
@@ -1242,14 +1242,15 @@ def go_repo(module: str, version:str='', download:str=None, name:str=None, insta
     requirements = " ".join(requirements)
 
     build_tag_args = '--build_tag ' + ' --build_tag '.join(build_tags) if build_tags else ''
+    label_args = '--label ' + ' --label '.join(labels) if labels else ''
```
Comment (toastwaffle) [inline build_defs/go.build_defs:1245]:
```suggestion
    label_args = [f"--label {label}" for label in labels].join(" ")
```

Comment (toastwaffle) [reply build_defs/go.build_defs:1245]: I have to think really hard about the precedence to the ternary operation to be convinced that this works as intended; I think this is more clear. I would change the line above to match as well.

File: tools/please_go/generate/generate.go [hunk 2]
```diff
@@ -30,9 +30,10 @@ type Generate struct {
 	knownImportTargets map[string]string // cache these so we don't end up looping over all the modules for every import
 	thirdPartyFolder   string
 	install            []string
+	extraLabels        []string
```
Comment (toastwaffle) [inline tools/please_go/generate/generate.go:33]: why is this called `extraLabels` and not just `labels`?

File: build_defs/go.build_defs (hunk 1 above)
Comment (mspraggs) [reply build_defs/go.build_defs:1245]: Sure, makes sense.

File: tools/please_go/generate/generate.go (hunk 2 above)
Comment (mspraggs) [reply tools/please_go/generate/generate.go:33]: I wasn't sure whether there were a set of base labels that would be applied to the target besides these. I will rename if not.

**Review by toastwaffle — APPROVED** [review]

**Pull request merged**